subxt/src/backend/unstable/follow_stream_unpin.rs

```diff
@@ -114,14 +115,23 @@ impl<Hash: BlockHash> Stream for FollowStreamUnpin<Hash> {
                 }
                 FollowStreamMsg::Event(FollowEvent::Initialized(details)) => {
                     // The first finalized block gets the starting block_num.
-                    let rel_block_num = this.rel_block_num;
-                    // Pin this block, but note that it can be unpinned any time since it won't show up again (except
-                    // as a parent block, which we are ignoring at the moment).
-                    let block_ref =
-                        this.pin_unpinnable_block_at(rel_block_num, details.finalized_block_hash);
+                    let mut rel_block_num = this.rel_block_num;
+
+                    let mut finalized_block_hashes =
+                        Vec::with_capacity(details.finalized_block_hashes.len());
+
+                    for finalized_block in &details.finalized_block_hashes {
+                        // Pin this block, but note that it can be unpinned any time since it won't show up again (except
+                        // as a parent block, which we are ignoring at the moment).
+                        let block_ref =
+                            this.pin_unpinnable_block_at(rel_block_num, *finalized_block);
+
+                        finalized_block_hashes.push(block_ref);
+                        rel_block_num += 1;
```

Comment:
nit: would `this.rel_block += 1` work here?

Reply:
Good spot; now there are multiple we should increment `this.rel_block_num` here!

Reply:
the old code didn't increment that but sure lemme fix that..